docs/installation.rst

```diff
@@ -123,38 +123,75 @@ For example, on Debian-based Linux we need to also do:
 
-.. code::
+- ``Dockerfile.devenv`` - This is the ``Dockerfile`` used to build the development image (``ghcr.io/nextsimhub/nextsimdg-dev-env``), that is
+  used in the GitHub CI.
+- ``Dockerfile.production`` - This ``Dockerfile`` is based off of the development image and it
+  additionally installs ``nextsim`` so that you can run on any machine with ``docker`` installed.
+
+Development Dockerfile
+^^^^^^^^^^^^^^^^^^^^^^
+
+A development image is provided on the nextsimhub `GitHub container registry
+<https://github.com/orgs/nextsimhub/packages>`_ because it is needed for the CI.
+
+If in future, this needs to be replaced. Please see `instructions
+<https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry>`_
+on the GitHub website.
+
+To build the docker image, please use these instructions:
+
+.. code-block:: console
+
+    docker build --file Dockerfile.devenv . -t ghcr.io/nextsimhub/nextsimdg-dev-env:latest
+
+.. note::
+   The formatting of the image name **is important**. The format is
+   ghcr.io/NAME_OF_REPOSITORY/NAME_OF_IMAGE:TAG
+
+If you want to test or use the image locally, use the following command:
+
+.. code-block:: console
+
+   docker pull ghcr.io/nextsimhub/nextsimdg-dev-env:latest
+
+Production Dockerfile
+^^^^^^^^^^^^^^^^^^^^^
+
+The production image is not stored on the nextsimhub `GitHub container registry
+<https://github.com/orgs/nextsimhub/packages>`_ because it is not needed for the CI. Users of the
+code may be interested in building their own. The instructions are as follows:
+
+.. code-block:: console
+
+   docker build --file Dockerfiles/Dockerfile.production . -t nextsim-production:latest
+
+This will build a local image of the nextsim code. The production ``Dockerfile`` supports additional
+build arguments (``--build-arg`` that can be specified at build time). For example, to build with
+``MPI`` enabled, using 4 processes to compile, use the following command,
+
+.. code-block:: console
+
+   docker build --file Dockerfiles/Dockerfile.production --build-arg mpi=ON --build-arg jobs=4 . -t nextsim-production:latest
 
-        docker build . --file Dockerfiles/Dockerfile_Ubuntu -t nextsim
-        docker run -it --entrypoint bash nextsim
-        
-The Ubuntu Dockerfiles create a full Ubuntu OS environment, copy over the content of the directory it is run from (which should be your nextsim root directory) and install and build nextsim in there for you. That way, you can easily try out your current code changes.
-
-With the last command, you are inside your running container on the bash and can use your usual command line commands to look around and run the code.
-    
-    
-.. _NetCDF: https://www.unidata.ucar.edu/software/netcdf/
-.. _Boost: https://www.boost.org/
-.. _CMake: https://cmake.org/
-.. _Homebrew: https://brew.sh/
-.. _Docker: https://www.docker.com/
+For a full list of options, please see ``Dockerfile.production``. By default ``MPI`` and ``xios``
```

Comment:
It would be nice to have the ``Dockerfile.production`` here be a hyperlink to the relevant file on GitHub, especially since it says "see" this file.

Reply:
I had a thought about this. I can't think of a sensible way to do this. If it was a `README.md` I would link to the relative path. Because it's external I would need a URL. But if I do that it will have to link to a specific branch/verison/commit hash. But then it would be out of date after any commit/PR that changes the file. Am I missing something?

Reply:
Yeah I suppose you're right, especially if it would have to go via `develop` before `main`.